### PR TITLE
ConnectionException must be notified through the sourceCallback

### PR DESCRIPTION
**Pull request by jmbarbosa80**

### CODE REVIEW

File: modules/ROOT/pages/reconnection.adoc [hunk 1]
```diff
@@ -148,7 +148,7 @@ Connection connection = connectionProvider.connect();
   try {
     connection.startStreaming();
   } catch(Exception e){
-    throw new ConnectionException(e, connection); //<1>
```
Comment (ndinu) [inline modules/ROOT/pages/reconnection.adoc:151]: We should also modify the text that is above, which states that the exception must be thrown just like in operations.

**Review by ndinu — DISMISSED** [review]

**Review by jmbarbosa80 — COMMENTED** [review]
LGTM

File: modules/ROOT/pages/reconnection.adoc [hunk 2]
```diff
@@ -133,11 +133,10 @@ could make the reconnection mechanism fail to work properly.
 [[starting-source]]
 ==== Starting Source
 
-When starting a Source, the execution of the `onStart()` method, if a connection
-problem is found and is required to be raised, in the same way as operations, a
-`ConnectionException()` must be thrown.
-Throwing the connection exception, the Source will be stopped, and will try to
-reconnect and start the Source again.
+A source is in the starting source stage when the onStart() method executes. If
```
Comment (dukesphere) [inline modules/ROOT/pages/reconnection.adoc]:
```suggestion
A source is in the starting source stage when the `onStart()` method executes. 
```

File: modules/ROOT/pages/reconnection.adoc [hunk 3]
```diff
@@ -133,11 +133,10 @@ could make the reconnection mechanism fail to work properly.
...
+a connection error exists, it must be communicated through the SourceCallback
```
Comment (dukesphere) [inline modules/ROOT/pages/reconnection.adoc]:
```suggestion
Connection errors must be communicated through the `SourceCallback`
```

File: modules/ROOT/pages/reconnection.adoc [hunk 4]
```diff
@@ -133,11 +133,10 @@ could make the reconnection mechanism fail to work properly.
...
+using the onConnectionException() method. The source will stop and attempt to
```
Comment (dukesphere) [inline modules/ROOT/pages/reconnection.adoc]:
```suggestion
using the `onConnectionException()` method. The source stops and attempts to
```

**Review by dukesphere — COMMENTED** [review]
mainly adding backticks for code-like formatting

**Review by dukesphere — APPROVED** [review]

**Pull request merged**